TEST API without signup & restore password

ShahRustam opened the pull request:
1)test login GET
2)test items POST
3)test item(add) POST
4)test item/<id> PUT
5)test item/<id> DELETE
5)test logout POST

ShahRustam committed:
TEST API without signup & restore password

1)test login GET
2)test items POST
3)test item(add) POST
4)test item/<id> PUT
5)test item/<id> DELETE
5)test logout POST

AlexVegner commented on test.py:
```diff
@@ -0,0 +1,58 @@
+class TestServerFunctionality(unittest.TestCase):
+
+	def setUp(self):
+		self.url= 'http://ec2-52-11-248-121.us-west-2.compute.amazonaws.com/api/v1'
```
def setUp(self): 
It will be executed before each test, so it's not correct place for constant
move it to
+class TestServerFunctionality(unittest.TestCase):
     BASE_URL = 'http://ec2-52-11-248-121.us-west-2.compute.amazonaws.com/api/v1'

also yo can move it to another class BASE_URL vill be used in different places of code

ShahRustam replied:
setUp function will be run before each test, a built-in method of unittest.

AlexVegner commented on test.py:
```diff
@@ -0,0 +1,58 @@
+
+	def test_1Login(self):
+		requestLogin = requests.post(self.url+'/login', json= self.loginval)
+		Token=(requestLogin.json().get('data')).get('tokenKey')
```
1. variables should start from lover case character should be token 
2. use space between variable and expression token = (req...

AlexVegner commented on test.py:
```diff
@@ -0,0 +1,58 @@
+		requestLogin = requests.post(self.url+'/login', json= self.loginval)
+		Token=(requestLogin.json().get('data')).get('tokenKey')
+		self.assertEqual(requestLogin.json().get('status'), 1)
+		EmailVerif= (requestLogin.json().get('data')).get('emailVerified')
```
emailVerif = (requ....

AlexVegner commented on test.py:
```diff
@@ -0,0 +1,58 @@
+		self.assertEqual(requestLogin.json().get('status'), 1)
+		EmailVerif= (requestLogin.json().get('data')).get('emailVerified')
+		self.assertEqual(EmailVerif,1)
+		userid= (requestLogin.json().get('data')).get('user_id')
```
userid = (r...

AlexVegner commented on test.py:
```diff
@@ -0,0 +1,58 @@
+		self.assertEqual(EmailVerif,1)
+		userid= (requestLogin.json().get('data')).get('user_id')
+		self.assertEqual(EmailVerif,1)
+		return Token
```
token

AlexVegner commented on test.py:
```diff
@@ -0,0 +1,58 @@
+		self.url= 'http://ec2-52-11-248-121.us-west-2.compute.amazonaws.com/api/v1'
+		self.loginval= {"login":"test", "password":"test"}
+
+	def test_1Login(self):
```
1. try to define test name more readable  test_login_positive
2. if you using _ in method name then all character of word should be with lover case word_word_word for tests we using it. Or camel case wordWordWord for ordinary methods.

AlexVegner commented on test.py:
```diff
@@ -0,0 +1,58 @@
+		requestItems = requests.get(self.url+'/items', headers=headers)
+		self.assertEqual(requestItems.json().get('status'), 1)
+		todoData= (requestItems.json().get('data')).get('todoData')
+		for x in range(0,len(todoData)):
```
Test should be independent, if you get and check data then you need to set it for this test, or other test can corrupt work of this test

AlexVegner commented on test.py:
```diff
@@ -0,0 +1,58 @@
+		self.assertEqual(EmailVerif,1)
+		return Token
+
+	def test_2Items(self):
```
bad naming test_get_items

AlexVegner commented on test.py:
```diff
@@ -0,0 +1,58 @@
+			self.assertEqual(todoData[x].get('id'),(x+1))
+
+
+	def test_3ItemAdd(self):
```
bad name

AlexVegner commented on test.py:
```diff
@@ -0,0 +1,58 @@
+		return Token
+
+	def test_2Items(self):
+		Token= self.test_1Login()
```
use class field for token

``` python
+class TestServerFunctionality(unittest.TestCase):
     token
```

or use specific function: 

``` python
def test_3ItemAdd(self):

     if self.toten:
        return self.toten
    else:
        login code....  
        self.toten = ...
        return self.toten
```

bad naming def test_3ItemAdd(self):

AlexVegner commented on test.py:
```diff
@@ -0,0 +1,58 @@
+
+
+	def test_3ItemAdd(self):
+		Token= self.test_1Login()
```
use one voice stile for your code 
" = "

AlexVegner replied:
use one voice stile for your code 
" = "

AlexVegner commented on test.py:
```diff
@@ -0,0 +1,58 @@
+		requestItem = requests.post(self.url+'/item', json= {"title":"test4", "description":"test4"}, headers=headers)
+		self.assertEqual(requestItem.json().get('status'), 1)
+
+	def test_4ItemEdit(self):
```
bad naming

AlexVegner commented on test.py:
```diff
@@ -0,0 +1,58 @@
+		requestItemEdit = requests.put(self.url+'/item/4', json= {"title":"test4", "description":"test4"}, headers=headers)
+		self.assertEqual(requestItemEdit.json().get('status'), 1)
+
+	def test_5DeleteItem(self):
```
bad name

AlexVegner commented on test.py:
```diff
@@ -0,0 +1,58 @@
+		requestItemDelete = requests.delete(self.url+'/item/4', headers=headers)
+		self.assertEqual(requestItemDelete.json().get('status'), 1)
+
+	def test_6Logout(self):
```
bad name

AlexVegner commented:
Please split tests, each api should have own file with class, all of them should be independent from other.

ShahRustam committed:
TEST API without signup & restore password (fixed)

1)test login GET
2)test items POST
3)test item(add) POST
4)test item/ PUT
5)test item/ DELETE
5)test logout POST

ShahRustam commented:
Tests runs in alphabetical order. I put in the names of the numbers of tests.

ShahRustam committed:
TEST API

1)test login GET
2)test items POST
3)test item(add) POST
4)test item/ PUT
5)test item/ DELETE
6)test logout POST
7)test signup POST
8)test restore password POST

ShahRustam committed:
update
